### PR TITLE
[gen3][esomx] Define E404X platform

### PR DESCRIPTION
**Pull request by scott-brust**
### Problem
E404X Hardware needs to be supported in DVOS

### Solution
Add new `PLATFORM_ID` for `esomx` platform, largely based off of boron module/platform definitions

### Steps to Test
Build using new platform and test with boron hardware:
```
make PLATFORM=esomx APP=tinker clean all
```

### Example App
Use tinker

---

### Completeness

- [x] User is totes amazing for contributing!
- [ ] Contributor has signed CLA ([Info here](https://github.com/spark/firmware/blob/develop/CONTRIBUTING.md))
- [ ] Problem and Solution clearly stated
- [ ] Run unit/integration/application tests on device
- [ ] Added documentation
- [ ] Added to CHANGELOG.md after merging (add links to docs and issues)

### CODE REVIEW

File: platform/MCU/nRF52840/inc/platform_config.h [hunk 1]
```diff
@@ -37,7 +37,8 @@
     PLATFORM_ID == PLATFORM_ASOM || \
     PLATFORM_ID == PLATFORM_BSOM || \
     PLATFORM_ID == PLATFORM_B5SOM || \
-    PLATFORM_ID == PLATFORM_TRACKER
+    PLATFORM_ID == PLATFORM_TRACKER || \
+    PLATFORM_ID == PLATFORM_ESOMX
```
Comment (XuGuohui) [inline platform/MCU/nRF52840/inc/platform_config.h:41]: If the E404X is gping to adopt an 8MB external flash, we need to modify line 52.

Comment (scott-brust) [reply platform/MCU/nRF52840/inc/platform_config.h:41]: I think there is still discussion about if E404X will support only 4mb filesystem or not, regardless of flash part. I updated `flash_mal` to support 8mb for now

**Comment by scott-brust** [comment]
> I think the following files need to be modified as well:
> 
> 1. platform/MCU/nRF52840/src/hw_config.c, line 197
> 2. user/applications/tinker/application.cpp
> 3. user/tests/wiring/no_fixture_long_running/network.cpp, pwm.cpp
> 4. user/tests/wiring/spi_master_slave

I'm working on pinmap next and will update tinker and tests then

File: hal/src/boron/hal_platform_config.h [hunk 2]
```diff
@@ -28,6 +28,18 @@
...
+
+#define HAL_PLATFORM_POWER_MANAGEMENT (1)
+#define HAL_PLATFORM_PMIC_BQ24195 (1)
+#define HAL_PLATFORM_PMIC_BQ24195_FAULT_COUNT_THRESHOLD (4)
+#define HAL_PLATFORM_FUELGAUGE_MAX17043 (1)
+#define HAL_PLATFORM_PMIC_BQ24195_I2C (HAL_I2C_INTERFACE2)
+#define HAL_PLATFORM_FUELGAUGE_MAX17043_I2C (HAL_I2C_INTERFACE2)
+#define HAL_PLATFORM_RADIO_ANTENNA_INTERNAL (0)
+#define HAL_PLATFORM_RADIO_ANTENNA_EXTERNAL (1)
+#define HAL_PLATFORM_FLASH_MX25L3233F (1)
```
Comment (artyLee) [inline hal/src/boron/hal_platform_config.h:41]: If there is using 8M flash, need to change macro `#define HAL_PLATFORM_FLASH_MX25L3233F (1)` to `#define HAL_PLATFORM_FLASH_MX25R6435FZNIL0 (1)`

File: hal/src/boron/pinmap_hal.c [hunk 3]
```diff
@@ -123,6 +123,58 @@ const uint8_t NRF_PIN_LOOKUP_TABLE[48] = {
 
 #endif // PLATFORM_ID == PLATFORM_BSOM
 
+#if PLATFORM_ID == PLATFORM_ESOMX
+
+static Hal_Pin_Info s_pin_map[TOTAL_PINS] = {
```
Comment (artyLee) [inline hal/src/boron/pinmap_hal.c:128]: This pin map looks not match our Evo board schematic. Maybe you will change it later. Please reference: [evo schematic](https://github.com/particle-iot/Evo/tree/main/Eagle/v0.02)

File: hal/src/boron/pinmap_defines.h [hunk 4]
```diff
@@ -167,3 +167,75 @@
 #define UBVINT 37
 
 #endif // PLATFORM_ID == PLATFORM_BSOM
+
+#if PLATFORM_ID == PLATFORM_ESOMX
```
Comment (artyLee) [inline hal/src/boron/pinmap_defines.h:171]: Synchronization update from pin map.

File: platform/MCU/nRF52840/inc/platform_config.h (hunk 1 above)
Comment (scott-brust) [reply platform/MCU/nRF52840/inc/platform_config.h:41]: E404X will support 8mb flash, and expose 4mb to user as FS, like tracker and b5som here.

**Review by XuGuohui — APPROVED** [review]

**Pull request merged**